app/lib/file_parse_service.rb

```diff
@@ -24,19 +24,23 @@ def self.run_parse_job(study_file, study, user, reparse: false, persist_on_fail:
           error: "File: #{study_file.upload_file_name} is already parsing"
       }
     else
+      self.create_bundle_from_file_options(study_file, study)
       case study_file.file_type
       when 'Cluster'
         job = IngestJob.new(study: study, study_file: study_file, user: user, action: :ingest_cluster, reparse: reparse,
                             persist_on_fail: persist_on_fail)
         job.delay.push_remote_and_launch_ingest
+        # check if there is a coordinate label file waiting to be parsed
+        # must reload study_file object as associations have possibly been updated
+        study_file.reload
+        if study_file.study_file_bundle.try(:completed?)
+          study_file.bundled_files.each do |coordinate_file|
+            run_at = 1.minutes.from_now # give cluster parse job time to initiate
```

Comment:
How reliable is this hard-coded time?  Would the cluster parse job ever now initiate within a minute?

Not a blocking concern; just a reflexive question I'd ask for almost any `sleep`-like code.

Reply:
This is just to avoid a race condition where the coordinate file parse can start before the parent cluster file job is fully invoked.  If that happens, it will fail the coordinate file job because the cluster file isn't labeled as "parsing".  I could have done a manual update of the parent file there, but since a job is about to be launched via `IngestJob`, it seemed cleaner/easier to just wait a little and allow the normal operations to take care of themselves.  So to answer the question, unless something totally unforeseen happens, 1 minute is more than enough time to wait (a few seconds is really all that's needed).

Reply:
b46fb50